### PR TITLE
Migrate to solana-keygen

### LINKED ISSUE #593: Add solana-keygen
The problem:

```bash
solana-mint > id.json
solana-wallet airdrop --tokens=123 --mint=id.json
solana-fullnode-config > validator.json     <---- doh! not the keypair from id.json
```

Related mess: We're using `solana-mint` as identity files because it happens to generate new keypairs. Then in `solana-wallet`, we call the parameter `--mint` because it's in the Mint JSON format and otherwise has no relationship to the Mint's file format.

There should be no `solana-mint` and instead just a `solana-keygen`. To get a token count to `solana-genesis`, we'd pass that in as a CLI option instead of through `mint.json`.

By default, `solana-keygen` should create `~/.config/solana/id.json` and include a CLI option to change that path. The format of the file could either be `Vec[u8]` or a base58-encoded string.

Likewise, by default, `solana-wallet` and `solana-fullnode` should look for `id.json` in the home directories. `solana-genesis`, on the other hand, should point to a different location explicitly for the Mint's identity file.

Here's what the new workflow would look like:

```bash
solana-keygen
solana-wallet airdrop --tokens=123
solana-fullnode-config > validator.json     <---- w00t! the keypair from ~/.config/solana/id.json
```

And for genesis:

```bash
solana-keygen -o - | solana-genesis -o genesis.log
```

### PR DESCRIPTION
**Pull request by garious**
Fixes #593

A bunch of code lifted from #595

### CODE REVIEW

**Comment by garious** [comment]
This is busted. More to come...

**Comment by CriesofCarrots** [comment]
I have have wallet.rs updated. I will copy a couple things out of this (namely read_keypair function).

**Comment by mvines** [comment]
lmk when this is ready for review

**Comment by garious** [comment]
@mvines, @CriesofCarrots, this is ready for review now.

File: src/bin/keygen.rs [hunk 1]
```diff
@@ -1,14 +1,49 @@
...
 use ring::rand::SystemRandom;
 use ring::signature::Ed25519KeyPair;
+use std::env;
 use std::error;
+use std::fs::{self, File};
+use std::io::Write;
+use std::path::Path;
 
 fn main() -> Result<(), Box<error::Error>> {
+    let matches = App::new("solana-genesis")
```
Comment (rob-solana) [inline src/bin/keygen.rs]: is this name supposed to be solana-keypair?

File: src/bin/keygen.rs [hunk 2]
```diff
@@ -1,14 +1,49 @@
...
+        .arg(
+            Arg::with_name("outfile")
+                .short("o")
+                .long("outfile")
+                .value_name("PATH")
+                .takes_value(true)
+                .help("Number of tokens with which to initialize mint"),
```
Comment (rob-solana) [inline src/bin/keygen.rs]: this help looks wrong...

File: src/bin/keygen.rs (hunk 1 above)
Comment (garious) [reply src/bin/keygen.rs]: solana-keygen, yes. fixed

File: src/bin/keygen.rs (hunk 2 above)
Comment (garious) [reply src/bin/keygen.rs]: yep, thanks! fixed

**Pull request merged**